### ISSUE TITLE
Add Rubio et al 2020 Schizosaccharomyces pombe example dataset 

### ISSUE THREAD

**Issue by 3mma-mack** [riboviz/example-datasets#69]
Add Rubio et al 2020 **Ribosome profiling reveals ribosome stalling on tryptophan codons and ribosome queuing upon oxidative stress in fission yeast**  as a new S. pombe example dataset.  This paper includes a range of stresses including heavy metal stress, oxidative stress, heat stress, osmotic stress and exposure to an alkylating agent. 

Link to [paper](https://academic.oup.com/nar/article/49/1/383/6031432)
links to data:
Response to cadmium sulphate - [E-MTAB-8746](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-8746/samples/)
Response to hydrogen peroxide - [E-MTAB-8744](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-8744/samples/)
Response to heat shock - [E-MTAB-8686](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-8686/samples/)
Response to sorbitol - [E-MTAB-8602](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-8602/samples/)
Response to methyl methanesulfonate - [E-MTAB-8745](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-8745/samples/)

**Comment by 3mma-mack** [comment]
Having done some digging I have found that the dataset includes Barcodes and UMIs, as shown in the file below, which is adapted from the supplementary data from [Duncan & Mata, 2014](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4082408/), as Rubio uses the methods described there. 

[Rubio et al barcodes.xlsx](https://github.com/riboviz/example-datasets/files/6196519/Rubio.et.al.barcodes.xlsx)

The format is 2nt of UMI, a 4nt barcode, then a 3 nt UMI, so I am going to have to spend some time figuring out what the regular expression would be but my current guess is it will be something like: ^(?P<umi_1>.{2})(?P<cell_1>.{4})(?P<umi_2>.{3})

**Comment by 3mma-mack** [comment]
A control sample from the cadmium sulphate experiment was run through riboviz with the following config file:

[Rubio_2020_stress_config.txt](https://github.com/riboviz/example-datasets/files/6278686/Rubio_2020_stress_config.txt)

This ran successfully, however the graphs produced look very wrong, with no periodicity and the majority of the few reads present being around 40 nt long. 
[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6278696/3nt_periodicity.pdf)
[pos_sp_rpf_norm_reads.pdf](https://github.com/riboviz/example-datasets/files/6278697/pos_sp_rpf_norm_reads.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6278699/read_lengths.pdf)
[startcodon_ribogrid.pdf](https://github.com/riboviz/example-datasets/files/6278701/startcodon_ribogrid.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6278703/startcodon_ribogridbar.pdf)

I believe this is due to either the adapters, barcodes or UMIs

**Comment by 3mma-mack** [comment]
Created the branch Rubio-Sp_2020-69 and added config file. In a meeting with Flic we noticed that the parameter 'Sample_sheet' was empty, so I added a barcodes TSV file and included a path to this file in the sample_sheet parameter

Am currently rerunning the dataset on riboviz

**Comment by 3mma-mack** [comment]
Following another unsuccessful run, I have had a dive into the logs, which I have copied and uploaded here. 
[command_log_cut_adapt.txt](https://github.com/riboviz/example-datasets/files/6286145/command_log_cut_adapt.txt)
[command_log_extract_umi.txt](https://github.com/riboviz/example-datasets/files/6286146/command_log_extract_umi.txt)
[command_log_hisatORF.txt](https://github.com/riboviz/example-datasets/files/6286152/command_log_hisatORF.txt)

I am not sure how to interpret the cutAdapters log, but the lines:
```
Total reads processed:              78,315,327
Reads with adapters:                17,243,079 (22.0%)
Reads written (passing filters):    78,315,327 (100.0%)
```
Suggests to me that the adapter may not be entirely correct.

At the bottom of the 'extractUmis' log the following line is present; 
```
2021-04-09 11:18:41,132 INFO Input Reads: 78315327
2021-04-09 11:18:41,132 INFO regex matches read1: 78315327
2021-04-09 11:18:41,132 INFO Reads output: 78315327
```
Which seems positive, and I think it means the regular expression is correct 
However what concerns me the most is that the HisatORF log states:

`0.04% overall alignment rate`

Which is definitely not good. @FlicAnderson what do you think?

**Comment by 3mma-mack** [comment]
> Having done some digging I have found that the dataset includes Barcodes and UMIs, as shown in the file below, which is adapted from the supplementary data from [Duncan & Mata, 2014](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4082408/), as Rubio uses the methods described there.
> 
> [Rubio et al barcodes.xlsx](https://github.com/riboviz/example-datasets/files/6196519/Rubio.et.al.barcodes.xlsx)
> 
> The format is 2nt of UMI, a 4nt barcode, then a 3 nt UMI, so I am going to have to spend some time figuring out what the regular expression would be but my current guess is it will be something like: ^(?P<umi_1>.{2})(?P<cell_1>.{4})(?P<umi_2>.{3})

'Rubio et al barcodes' shows that the overall sequence including UMIs, barcodes and adapters is: /5Phos/NNATTGNNNAGATCGGAAGAGCGTCGTGT/iSp18/CACTCA/iSp18/CTGAACCGCTCTTCCGATCTTTTTTTTTTTTTTTTTTTTTVN

In the previous runs I had been using `AGATCGGAAGAGCGTCGTGT` as the adapter sequence, which the cutAdapt log seemed to  show as being present in 22% of reads. This time I used `CTGAACCGCTCTTCCGATC` as the adapter, which the included cutAdapt log seems to show as being present in 30.6% of samples. 

Again, Hisat2ORF shows that only 0.03% of reads are being aligned. 

[command_cutAdapt.txt](https://github.com/riboviz/example-datasets/files/6310965/command_cutAdapt.txt)
[command_extractUMIs.txt](https://github.com/riboviz/example-datasets/files/6310966/command_extractUMIs.txt)
[command_Hisat2ORF.txt](https://github.com/riboviz/example-datasets/files/6310969/command_Hisat2ORF.txt)

**Comment by 3mma-mack** [comment]
I had another read of the [Rubio et al paper](https://academic.oup.com/nar/article/49/1/383/6031432) and decided to test out using the adapters and UMIs used in the paper referenced for data processing and read alignment methods, which is coincidentally the [Duncan & Mata paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5828588/) already included in example-datasets. 

Using the adapter sequence TGGAATTCTCGGGTGCCAAGG and UMI regular expression for 4 UMIs at either end of the sequence, the following graphs were created.

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6319623/3nt_periodicity.pdf)
[pos_sp_rpf_norm_reads.pdf](https://github.com/riboviz/example-datasets/files/6319624/pos_sp_rpf_norm_reads.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6319625/read_lengths.pdf)
[startcodon_ribogrid.pdf](https://github.com/riboviz/example-datasets/files/6319626/startcodon_ribogrid.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6319627/startcodon_ribogridbar.pdf)

These look **much** better, if a little messy. Moral of the story, make sure you use the methods paper to take adapters and UMIs from.

**Comment by FlicAnderson** [comment]
This does look much better! What are the alignment values like this time?

**Comment by 3mma-mack** [comment]
> This does look much better! What are the alignment values like this time?

It had a 49.38% overall alignment rate this time 

[command_Hisat2ORF_15_04.txt](https://github.com/riboviz/example-datasets/files/6323845/command_Hisat2ORF_15_04.txt)

**Comment by 3mma-mack** [comment]
I have updated the config files to include the correct adapter sequence, UMIs, samples available as comments and remove the path to barcodes. 

I am going to attempt a run with a lot of samples, though sample ERR3850619 is having some problems with SRAtoolkit, as it produces the error:
`
2021-04-16T09:21:40 fastq-dump.2.8.2 err: unknown while writing file within file system m error errno='Disk quota exceeded(122)'`

I saw the 'disk quota exceeded' part and cleaned up my scratch space however it continued to produce the error. Additionally sample ERR3850618 was successfully fetched and dumped following attempts to use fastq-dump on ERR3850618

**Comment by kavousan** [comment]
> I saw the 'disk quota exceeded' part and cleaned up my scratch space however it continued to produce the error.

Do we know if it is simple and free to get more disk space, and can we find out what runs out of space (`df -k` may be your friend)?

**Comment by 3mma-mack** [comment]
The issue was the folder ncbi, where the SRA download takes place, was in my home directory, which has limited space. I have moved this folder into my scratch directory , which has solved the problem and I am now able to download the samples again.

**Comment by ewallace** [comment]
Actually, there is a problem with overload of scratch space on Eddie. So instead we need to use a different (paid) group space for "scratch-style computation. But I don't remember the details.

It looks like we need to update [run-on-eddie.md](https://github.com/riboviz/riboviz/blob/develop/docs/user/run-on-eddie.md) to explain this better.

**Comment by ewallace** [comment]
@3mma-mack has run this ok on the 250nt fixed-buffer dataset, but has bugs on.
Expect to pull request and close in next week or so, @FlicAnderson will review.

**Comment by 3mma-mack** [comment]
I have double checked that this runs okay on multiple samples for the 250nt fixed buffer dataset. Overall all of the samples do seem to have far fewer reads than the Duncan & Mata dataset, and have a number of reads that are around 21nt in length, but this is consistent across the samples.


Heat shock: 

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371570/3nt_periodicity.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371571/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371572/startcodon_ribogridbar.pdf)


Heat shock control 

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371573/3nt_periodicity.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371574/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371575/startcodon_ribogridbar.pdf)


H2O2

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371576/3nt_periodicity.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371577/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371578/startcodon_ribogridbar.pdf)


H2O2 control

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371581/3nt_periodicity.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371582/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371584/startcodon_ribogridbar.pdf)


MMS

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371585/3nt_periodicity.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371587/startcodon_ribogridbar.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371588/read_lengths.pdf)


MMS control 

[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371591/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371592/startcodon_ribogridbar.pdf)
[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371589/3nt_periodicity.pdf)


Sorbitol 

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371593/3nt_periodicity.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371594/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371595/startcodon_ribogridbar.pdf)


Sorbitol control

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371596/3nt_periodicity.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371598/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371599/startcodon_ribogridbar.pdf)


Cadmium 

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371607/3nt_periodicity.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371610/startcodon_ribogridbar.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371609/read_lengths.pdf)


Cadmium control 

[3nt_periodicity.pdf](https://github.com/riboviz/example-datasets/files/6371612/3nt_periodicity.pdf)
[read_lengths.pdf](https://github.com/riboviz/example-datasets/files/6371613/read_lengths.pdf)
[startcodon_ribogridbar.pdf](https://github.com/riboviz/example-datasets/files/6371614/startcodon_ribogridbar.pdf)

**Comment by FlicAnderson** [comment]
Merged via #75
